common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -102,9 +102,14 @@ public void fetchBlocks(
     try {
       RetryingBlockFetcher.BlockFetchStarter blockFetchStarter =
           (blockIds1, listener1) -> {
-            TransportClient client = clientFactory.createClient(host, port);
-            new OneForOneBlockFetcher(client, appId, execId,
-              blockIds1, listener1, conf, downloadFileManager).start();
+            // Unless this client is closed.
```

Comment:
`this client` -> `this clientFactory`?

Reply:
`This client` here means ExternalBlockStoreClient, not the clientFactory.

Reply:
Does it match with the code `if (clientFactory != null) {`?

Reply:
I am fine for clientFactory, although I concern it sounds a bit not easy to understand, for end users.

Reply:
Changed to clientFactory.